lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -1,13 +1,32 @@
 module ActiveModelSerializers
   module Adapter
     class Attributes < Base
+      def initialize(*)
+        super
+        instance_options[:fieldset] ||= ActiveModel::Serializer::Fieldset.new(fields_to_fieldset(instance_options.delete(:fields)))
```

Comment:
the fieldset for the adapter is set at init. The fields the adapter is told to use is passed to serializable_hash. Not sure how good a distinction this is, but it's also how the json:api adapter works

Reply:
Since the JSON Adapter delegates to Attributes here, this is the only necessary adapter change

Reply:
honestly, this particular comment answered my long-standing question for this PR :smile: clever work :clap: